src/robot/running/context.py

```diff
@@ -39,12 +41,19 @@ def namespaces(self):
         return (context.namespace for context in self)
 
     def start_suite(self, suite, namespace, output, dry_run=False):
-        ctx = _ExecutionContext(suite, namespace, output, dry_run)
+        if self._contexts:
+            self.event_loop_ref = self._contexts[-1]._event_loop
+        ctx = _ExecutionContext(suite, namespace, output, dry_run, self.event_loop_ref)
```

Comment:
I believe the above could be written like this to avoid the need to have `self.event_loop_ref`:
```python
event_loop = self._contexts[-1]._event_loop if self._contexts else None
ctx = _ExecutionContext(..., event_loop)
```